lib/mavenlink/client.rb

```diff
@@ -70,9 +73,14 @@ def delete(path, arguments = {})
 
     # @return [Hash]
     def connection_options
+      if @custom_client_name?
+        @user_agent = "#{@custom_client_name} via Mavenlink Ruby Gem"
+      else
+        @user_agent = "Mavenlink Ruby Gem"
+      end
```

Comment:
If we let them override the user-agent, just let them specify the whole thing (dont append anything to their input), and only validate that it has `length > 1`